src/ripple/protocol/Indexes.h

```diff
@@ -89,6 +91,137 @@ getRippleStateIndex (Account const& a, Issue const& issue);
 
 uint256
 getSignerListIndex (Account const& account);
+
+//------------------------------------------------------------------------------
+
+/** A pair of SHAMap key and LedgerEntryType.
+    
+    A Keylet identifies both a key in the state map
+    and its ledger entry type.
+*/
+struct Keylet
+{
+    LedgerEntryType type;
+    uint256 key;
+
+    Keylet (LedgerEntryType type_,
+            uint256 const& key_)
+        : type(type_)
+        , key(key_)
+    {
+    }
+};
+
+/** Keylet computation funclets. */
+namespace keylet {
+
+/** Account root */
+struct account_t
+{
+    Keylet operator()(Account const& id) const;
+
+    // DEPRECATED
+    Keylet operator()(RippleAddress const& ra) const;
+};
+static account_t const account {};
+
+/** OWner directory */
+struct owndir_t
+{
+    Keylet operator()(Account const& id) const;
+};
+static owndir_t const ownerDir {};
+
+/** Skip list */
+struct skip_t
+{
+    Keylet operator()() const;
+
+    Keylet operator()(LedgerIndex ledger) const;
+};
+static skip_t const skip {};
+
+/** The amendment table */
+struct amendments_t
+{
+    Keylet operator()() const;
+};
+static amendments_t amendments {};
```

Comment:
`const`. If `const` isn't there gcc gives a warning about an unused variable. Interestingly, clang does not issue the warning.